supervisor/dbus/network/setting/generate.py

```diff
@@ -189,13 +202,16 @@ def get_connection_from_interface(
     if not uuid:
         uuid = str(uuid4())
 
+    llmnr = _map_mdns_setting(interface.llmnr)
+    mdns = _map_mdns_setting(interface.mdns)
+
     conn: dict[str, dict[str, Variant]] = {
         CONF_ATTR_CONNECTION: {
             CONF_ATTR_CONNECTION_ID: Variant("s", name),
             CONF_ATTR_CONNECTION_UUID: Variant("s", uuid),
             CONF_ATTR_CONNECTION_TYPE: Variant("s", iftype),
-            CONF_ATTR_CONNECTION_LLMNR: Variant("i", 2),
-            CONF_ATTR_CONNECTION_MDNS: Variant("i", 2),
+            CONF_ATTR_CONNECTION_LLMNR: Variant("i", llmnr),
+            CONF_ATTR_CONNECTION_MDNS: Variant("i", mdns),
```

Comment:
Variant probably can handle `IntEnum`'s, but better to be explicit:

```suggestion
            CONF_ATTR_CONNECTION_LLMNR: Variant("i", int(llmnr)),
            CONF_ATTR_CONNECTION_MDNS: Variant("i", int(mdns)),
```